frontend/src/app/components/modals/confirm-dialog/confirm-dialog.modal.ts

```diff
@@ -54,7 +56,7 @@ export class ConfirmDialogModal extends OpModalComponent {
 
   public confirmed = false;
 
-  private options:ConfirmDialogOptions;
+  public options:ConfirmDialogOptions;
```

Comment:
Instead of making it public, I would suggest to introduce a public variable `classes` and `passedData` and assign the values in the constructor like it is done below. Then you can also move [the check for the class](https://github.com/opf/openproject/pull/8420/files#diff-24b2f43c393e2d4847b3166d94380528R33) into the ts-code.